seed/views/v3/analyses.py

```diff
@@ -0,0 +1,74 @@
+# !/usr/bin/env python
+# encoding: utf-8
+"""
+:copyright (c) 2014 - 2020, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
+:author
+"""
+from drf_yasg.utils import swagger_auto_schema
+from django.http import JsonResponse
+from rest_framework import viewsets
+from rest_framework.status import HTTP_409_CONFLICT
+
+from seed.decorators import ajax_request_class, require_organization_id_class
+from seed.lib.superperms.orgs.decorators import has_perm_class
+from seed.models import Analysis
+from seed.serializers.analyses import AnalysisSerializer
+from seed.utils.api import api_endpoint_class
+from seed.utils.api_schema import AutoSchemaHelper
+
+
+class AnalysisViewSet(viewsets.ViewSet):
+    serializer_class = AnalysisSerializer
+    model = Analysis
+
+    @swagger_auto_schema(
+        manual_parameters=[
+            AutoSchemaHelper.query_org_id_field(True),
+            AutoSchemaHelper.query_integer_field('property_id', True, 'Property ID')
+        ]
+    )
+    @require_organization_id_class
+    @api_endpoint_class
+    @ajax_request_class
+    @has_perm_class('requires_member')
+    def list(self, request):
+        organization_id = request.query_params.get('organization_id', None)
+        property_id = request.query_params.get('property_id', None)
+        analyses = []
+        if property_id is not None:
+            analyses_queryset = Analysis.objects.filter(organization=organization_id, analysispropertyview__property=property_id).distinct()
+        else:
+            analyses_queryset = Analysis.objects.filter(organization=organization_id).distinct()
```

Comment:
I didn't have them at first but they were actually needed.  I can't remember the exact situation but one of the joins and filters resulted in there bing two identical analysis in the response.  I can go back and find the specific test that failed.

Reply:
I saw this note here: https://docs.djangoproject.com/en/3.1/ref/models/querysets/#django.db.models.query.QuerySet.distinct
> By default, a QuerySet will not eliminate duplicate rows. In practice, this is rarely a problem, because simple queries such as Blog.objects.all() don’t introduce the possibility of duplicate result rows. However, if your query spans multiple tables, it’s possible to get duplicate results when a QuerySet is evaluated. That’s when you’d use distinct()

Reply:
Right, and duplicates appearing would happen since we could have 2 AnalysisPropertyViews with the same property_id.

I think `distinct()` here makes sense in the case we're referencing the `analysispropertyview` relation, but maybe it doesn't make as much sense in the case where we're just using `organization_id`?

Reply:
True.  I'll take it off the second case.